bundles/framework/featuredata/plugin/FeatureDataPluginHandler.js

```diff
@@ -0,0 +1,72 @@
+import { StateHandler, controllerMixin } from 'oskari-ui/util';
+import { showFeatureDataFlyout } from './FeatureDataFlyout';
+
+class FeatureDataPluginUIHandler extends StateHandler {
+    constructor (plugin, mapModule, config) {
+        super();
+        this.setState({
+            activeTab: null
+        });
+        this.mapModule = mapModule;
+        this.addStateListener(() => this.updateFlyout());
+    }
+
+    getFeatureDataLayers () {
+        return Oskari.getSandbox()
+            .findAllSelectedMapLayers()
+            .filter(layer => layer.isVisibleOnMap() && layer.hasFeatureData && layer.hasFeatureData());
+    }
+
+    getFeaturesByLayerId (layerId) {
+        const featuresMap = this.mapModule.getVectorFeatures(null, { layers: [layerId] });
+        return featuresMap && featuresMap[layerId] ? featuresMap[layerId].features : null;
+    }
+
+    setActiveTab (layerId) {
+        const features = layerId ? this.getFeaturesByLayerId(layerId) : null;
+        this.updateState({
+            activeLayerId: layerId,
+            activeLayerFeatures: features
+        });
+    }
+
+    updateStateAfterMapEvent () {
+        const featureDataLayers = this.getFeatureDataLayers() || [];
+        const layerId = featureDataLayers?.find(layer => layer.getId() === this.state.activeLayerId) ? this.state.activeLayerId : featureDataLayers.map(layer => layer.getId())[0] || null;
+        const features = layerId ? this.getFeaturesByLayerId(layerId) : [];
+        this.updateState({
+            activeLayerId: layerId,
+            layers: featureDataLayers,
+            activeLayerFeatures: features
+        });
+    }
+
+    openFlyout () {
+        if (this.flyoutController) {
+            this.closeFlyout();
+            return;
+        }
+        const layers = this.getFeatureDataLayers() || null;
+        const activeLayerId = layers && layers.length ? layers[0].getId() : null;
+        const activeLayerFeatures = activeLayerId ? this.getFeaturesByLayerId(activeLayerId) : null;
+        this.state = { layers, activeLayerId, activeLayerFeatures };
+        this.flyoutController = showFeatureDataFlyout(this.state, this.getController());
```

Comment:
If we don't preload features to state on every map move even if the flyout is closed we should get them here. 

See if this kind of code works ok. It might not since the last line assumes that the updateState() has added the features BUT since in constructor the state changes trigger a call to updateFlyout() it might be ok :)

```suggestion
        const { layers, activeLayerId, activeLayerFeatures } = this.getState();
        if (!activeLayerFeatures) {
            // not empty features, but missing completely
            // empty should mean there is no features on viewport to list
            this.updateState({
                activeLayerFeatures: this.getFeaturesByLayerId(activeLayerId)
            });
        }
        this.flyoutController = showFeatureDataFlyout(this.getState(), this.getController());
```

Reply:
Didn't work out of the box but I implemented the basic idea like this.